src/bin/sol-fbp-generator/main.c

```diff
@@ -1057,6 +1183,14 @@ parse_args(int argc, char *argv[])
                 return false;
             }
             break;
+        case 'm':
+            if (access(optarg, R_OK) == -1) {
```

Comment:
I would prefer if the check is made when the file is actually used.

As there shouldn't be any serious consequences, I will leave this suggestion as optional.